cmd/cat-main.go

```diff
@@ -61,6 +66,55 @@ EXAMPLES:
 `,
 }
 
+// prettyStdout replaces some non printable characters
+// with <hex> format to be better viewable by the user
+type prettyStdout struct {
+	// Internal data to pretty-print
+	writer io.Writer
+	// Internal buffer which contains pretty printed
+	// form of binary (no printable) characters
+	buffer *bytes.Buffer
+}
+
+// newPrettyStdout returns an initialized prettyStdout struct
+func newPrettyStdout(w io.Writer) *prettyStdout {
+	return &prettyStdout{
+		writer: w,
+		buffer: bytes.NewBuffer([]byte{}),
+	}
+}
+
+// Read() returns pretty printed binary characters
+func (s prettyStdout) Write(input []byte) (int, error) {
+	inputLen := len(input)
+
+	// Convert no printable characters to '^?'
+	// and fill into s.buffer
+	for len(input) > 0 {
+		r, size := utf8.DecodeRune(input)
+		if unicode.IsPrint(r) || unicode.IsSpace(r) {
```

Comment:
we could simplify this to `unicode.IsPrint(r)` since `' '` is a printable character.

Reply:
> we could simplify this to unicode.IsPrint(r) since ' ' is a printable character.

yeah, true for space, but we need IsSpace() for other characters like '\n', '\r', '\t'

```
unicode.IsPrint('\n') => false
unicode.IsSpace('\n') => true
```